.github/workflows/pr-ci.yml

```diff
@@ -13,51 +49,55 @@ jobs:
-                                  dh-systemd \
-                                  fakeroot \
-                                  gcc \
-                                  git \
-                                  libnl-3-200 libnl-3-dev libnl-route-3-200 libnl-route-3-dev \
-                                  libnuma-dev \
-                                  libudev-dev \
-                                  uuid-dev \
-                                  make \
-                                  ninja-build \
-                                  pandoc \
-                                  pkg-config \
-                                  python \
-                                  rpm \
-                                  sparse \
-                                  valgrind \
-                                  wget
+          sudo apt-get install -y ${{ env.APT_PACKAGES }}
       - uses: actions/checkout@v2
       - name: Build Check
         run: |
-          git clone --depth 1 -b v34.1 https://github.com/linux-rdma/rdma-core.git
+          set -x
+          git clone --depth 1 -b ${{ env.RDMA_CORE_VERSION }} https://github.com/linux-rdma/rdma-core.git
           pushd rdma-core; bash build.sh; popd
-          export RDMA_CORE_PATH="rdma-core/build"
-          export LD_LIBRARY_PATH="$RDMA_CORE_PATH/lib:$LD_LIBRARY_PATH"
+          export LD_LIBRARY_PATH="${{ env.RDMA_CORE_PATH }}/lib:$LD_LIBRARY_PATH"
           ./autogen.sh
-          ./configure --prefix=$PWD/install --enable-efa=${RDMA_CORE_PATH} \
-                                            --enable-mrail \
-                                            --enable-psm3=${RDMA_CORE_PATH} \
-                                            --enable-rxd \
-                                            --enable-rxm \
-                                            --enable-shm \
-                                            --enable-tcp \
-                                            --enable-udp \
-                                            --enable-usnic \
-                                            --enable-verbs=${RDMA_CORE_PATH} \
+          ./configure --prefix=$PWD/install ${{ env.OFI_PROVIDER_FLAGS }} CC=${{ matrix.cc }}
+          make -j 2; make install
+          $PWD/install/bin/fi_info -l
+  hmem:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install dependencies (Linux)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y ${{ env.APT_PACKAGES }}
+      - name: Install CUDA
+        run: |
+          sudo apt-get install -y nvidia-cuda-toolkit
+      - name: Install ROCm
+        run: |
+          echo "Installing ROCm SDK"
+          # TODO: Install ROCm dependencies and add --with-rocm to build in next step
+      - name: Install Xe
+        run: |
+          echo "Installing Ze SDK"
+          # TODO: Install Ze dependencies and add --with-ze to build in next step
+      - uses: actions/checkout@v2
+      - name: HMEM Checks
```

Comment:
Should we separate the build and tests into different steps?

- HMEM Build
- HMEM Check (fi_info)
- HMEM Check (future ...)

Reply:
Sure, will get that in the next round when the Ze/ROCm stuff land. @aingerson I'm not sure how to install Ze stuff. I left a TODO note in the config for now. Will you or someone on your team be able to fill that in (and add in --with-ze to the hmem build)?

Reply:
@rajachan I'll make a note to add this once the ZE stuff is available publicly. Thanks!

Reply:
Here is the public info for Ze* (Level Zero).
https://dgpu-docs.intel.com/installation-guides/ubuntu/ubuntu-focal.html
```
- name: Install Xe
        run: |
          echo "Installing Ze SDK"
          sudo apt-get install -y gpg-agent wget
          wget -qO - https://repositories.intel.com/graphics/intel-graphics.key | sudo apt-key add -
          sudo apt-add-repository 'deb [arch=amd64] https://repositories.intel.com/graphics/ubuntu focal main'
          sudo apt-get update
          sudo apt-get install -y level-zero level-zero-dev
```
```
  --with-ze (default install is /usr/include/  "level_zero/ze_api.h")
```

Reply:
Here is the info for ROCm: https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html#ubuntu

Reply:
@acgoldma Nice, I didn't realize it was public yet. Go ahead and add them if you feel comfortable doing so. Otherwise, I can do it. Thanks!

Reply:
@aingerson, you can go ahead.